src/scripts/exclusion_term_expansion.py

```diff
@@ -51,19 +51,29 @@ class having `True` for `exclude_children`, but the child class has `False`.
 
 
 # Functions
-# TODO: use: https://bioregistry.readthedocs.io/
-#  en/latest/api/bioregistry.curie_from_iri.html?highlight=get_curie#bioregistry.curie_from_iri
-def uri_to_curie(uri: str) -> str:
-    """Takes an ontological URI and returns a CURIE. Works on the following patterns:
-    - http://.../PREFIX/CODE
-    - http://.../PREFIX_CODE"""
-    uri_list: List[str] = uri.split('/')
-    end_bit = uri_list[-1]
-    if '_' in end_bit:
-        curie = end_bit.replace('_', ':')
+# TODO: evaluate new bioregistry option https://github.com/biopragmatics/bioregistry/issues/480#issuecomment-1199235747
+def uri_to_curie(uri: str, prefix_map: Dict[str, str] = None) -> str:
+    """Takes an ontological URI and returns a CURIE. Works on the following patterns
+    # todo: defaulting prefix_map to None is temp for now but need to change to oak anyway soon
+    # todo: Use this solution until I get OAK to work for this use case"""
+    obo_purl = "http://purl.obolibrary.org/obo/"
+    curie = uri
+    uri = curie.replace('<', '').replace('>', '')
+    if prefix_map:
+        for k, v in prefix_map.items():
+            if uri.startswith(v):
+                curie = uri.replace(v, f"{k}:")
+        if uri.startswith(obo_purl):
+            uri = uri.replace(obo_purl, "")
+            curie = uri.replace("_", ":")
     else:
```

Comment:
I would prefer not to have a default for when no prefix_map is provided. just let it fail hard in this case, rows 70-76 are very error prone guesswork and you can imagine dozens of cases where it is simply wrong

Reply:
That suggestion makes sense. I think there might have been some cases in the past where I didn't have a `prefix_map`, but that's not a situation we want to be in anyway, and I think I've gotten past that point too, so I can remove it.

Reply:
done!